src/main/java/org/opensearch/geospatial/constants/QueryPreference.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.constants;
+
+/**
+ * Collection of query preference
+ */
+public class QueryPreference {
```

Comment:
Why do we need a new class, I think we can use existing [enum for Preference](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/routing/Preference.java#L39) here?